Update userguide (#155)
AY2324S2-CS2103T-T13-2/tp: docs/UserGuide.md

  * [Clearing All Entries](#clearing-all-entries--clear)
  * [Exiting Program](#exiting-the-program--exit)
  * [Saving Data](#saving-the-data)
  * [Editing Data File](#editing-the-data-file)
* [Future Enhancements](#future-enhancements-hammer)
* [Frequently Asked Questions](#faq-questionquestionquestion)
* [Known Issues](#known-issues-construction)
* [Command Summary](#command-summary)

--------------------------------------------------------------------------------------------------------------------

## Getting Started!

> **IMPORTANT**
> Ensure you have Java `11`  installed in your computer. If you do not have it installed, download it from [here.](https://www.oracle.com/sg/java/technologies/javase/jdk11-archive-downloads.html)
> Not sure if you have Java `11` downloaded, here's how to [check](#faq-questionquestionquestion).
1. Download the latest `internBook.jar` from [here](https://github.com/AY2324S2-CS2103T-T13-2/tp/releases).

2. Copy the file to the folder you want to use as the _home folder_ for your InternBook. 
>:bulb: **TIP**: Create a folder with internBook.jar so that the data files created will be stored in the same folder.

> 3. Open a command terminal, `cd` into the folder you put the jar file in, and use the `java -jar internBook.jar` command to run the application.

Reviewer: Maybe can link this to the FAQ on how to launch the app, or perhaps can put it here as well as an extra information

* Extraneous parameters for commands that do not take in inputs (such as `help`, `list`, `exit` and `clear`) will be ignored.<br>
  e.g. if the command specifies `help 123`, it will be interpreted as `help`.

* :exclamation: **Important:** Dates will only show in the GUI if both start date and end date are present.<br>

* If you are using a PDF version of this document, be careful when copying and pasting commands that span multiple lines as space characters surrounding line-breaks may be omitted when copied over to the application.
</box>

### Viewing help : `help`

Shows a pop-up window of our user guide for easier access.

![help message](images/help.png)

> Format: `help`

Author: the line is already before the image.

1. Input: `add -n Tik_Tok -p 91234-567 -e tiktokexample -t Soft-ware Engineer -d1 15-04-2024 -d2 2024-03-14`<br>
   * Error: Name cannot contain "_".
   ![name error message](images/name_error_msg.png)
   * Correction: Remove "_".
2. Input: `add -n TikTok -p 91234-567 -e tiktokexample -t Soft-ware Engineer -d1 15-04-2024 -d2 2024-03-14`<br>
   * Error: Phone number should only contain numbers.
   ![phone error message](images/phone_error_msg.png)
   * Correction: Remove non-numeric characters.
3. Input: `add -n TikTok -p 91234567 -e tiktokexample -t Soft-ware Engineer -d1 15-04-2024 -d2 2024-03-14`<br>
   * Error: Date format should be YYYY-MM-DD.
   * Correction: Adjust date format.
4. Input: `add -n TikTok -p 91234567 -e tiktokexample -t Soft-ware Engineer -d1 2024-04-15 -d2 2024-03-14`<br>
   * Error: Start date should precede end date.
   ![date error message](images/startDate_endDate_error.png)
   * Correction: Swap date order.
5. Input: `add -n TikTok -p 91234567 -e tiktokexample -t Soft-ware Engineer -d1 2024-04-14 -d2 2024-03-15`<br>
   * Error: Incorrect email format.
   ![email error message](images/email_error.png)

>    For the complete error message, please consult the "Remarks" column in the [Adding a Company](#adding-a-company-add) table.

Author: where did you see this back tracking?

   * **Before:**
   ![result for 'pre-editing'](images/pre_edit_name.png)
   *  **After:**
   ![result for 'post-editing'](images/post_edit_name.png)
2. Input: `edit 1 -e alphabet@example.com`
   * **Before:**
   ![result for 'pre-editing'](images/pre_edit_email.png)
   * **After:**
   ![result for 'post-editing'](images/post_edit_email.png)
3. Input: `edit 1 -t UI/UX Designer`
   * **Before:**
   ![result for 'pre-editing'](images/pre_edit_tag.png)
   * **After:**
   ![result for 'post-editing'](images/post_edit_tag.png)
4. Input: `edit 1 -p 91234567`
    * **Before:**
      ![result for 'pre-editing'](images/pre_edit_phone.png)
    * **After:**
      ![result for 'post-editing'](images/post_edit_phone.png)
5. Input: `edit 1 -d1 2024-04-15 -d2 2024-05-01`
    * **Before:**
      ![result for 'pre-editing'](images/pre_edit_dates.png)
    * **After:**
      ![result for 'post-editing'](images/post_edit_dates.png)

>:bulb: **TIP**:
> When editing the dates of an application, it is recommended to edit both start and end dates. 
> If only one of the date is edited, and the record is not showing anything, please refer to our [FAQ](#faq-questionquestionquestion).

> ### Locating companies by name: `find` :mag:

Reviewer: should we change the name of the title since now we are searching for tags as well? maybe it can be something like  Locating companies and tags by keywords

* Only the name and tags are searched.
* The result screen will display all the **tags and names** that match the keyword.
e.g. `Software` will return a Company with name `Software XYZ` as well as a Company with tag `Software Engineer`.
* Only Companies/Tags or its words **beginning with** the keyword will be returned.
    e.g. `Test` will return `Test Engineer` or `QA Tester`, but not `QATester`.

> * If there is a space in the keyword, it searches for a substring match with the whole word

Reviewer: i think theres a missing "that" in the middle of substring and match

> * `list` followed by `delete 2` deletes the 2nd company in the intern book.

Reviewer: maybe can follow the mark and unmark feature to make it more consistent, perhaps can include listed intern book, and if we are putting list first, we should do the same for mark and unmark as well

Also, maybe we can add the error handling of the index, similar to the mark and unmark feature as well, where it states that if the index is out of bound, an error message will be displayed

>   Examples:

Reviewer: i think the formatting of this can be shifted to the left to aligned with the rest of the UG to maintain consistency

* `sort a`
* `sort s`
* `sort e`

![sort example](images/sort.png)

:information_source: Remarks:

> - After sorting by start date or end date, you noticed that application that does not have their dates displayed is in 
> between applications with their dates displayed, please refer to our [FAQ](#faq-questionquestionquestion).

Reviewer 2: ```suggestion
- After sorting by start date or end date, if you notice that there are companies without dates displayed interspersed  
between companies with their dates displayed, please refer to our [FAQ](#faq-questionquestionquestion).
```

> - After adding an application or editing the dates of an application, the sort command has to be run again,

Reviewer: i think not necessarily dates, since can edit the name as well and can sort by name. Also i think we need to standardize the adding of an application OR adding of a company, because you can add the company in but havent applied yet. and at the top in the addcommand and those commands at the top it was about adding companies.

maybe can state that when adding a new company it will be added to the bottom of the list, and as for editing the arrangement will stay the same

* `reminder -r 7`
* `reminder -r off`

Steps to set a reminder:
* Input: `reminder -r 30`
![Reminder success](images/reminder_successful.png)
* The next time the app is launched, the reminder window will appear.
![Reminder example](images/reminder_example.png)
* If there are no applications outstanding, the reminder window below is shown.
![No reminder example](images/no_reminder_tasks.png)
* To switch off, simply input `reminder -r off`.

>:bulb:**TIP**:

> > To switch off reminder, simply key `reminder -r off`. 

Reviewer 2: These two points are identical so maybe we can remove one of them

**A**: For MAC users, open up your **Terminal**, and for Windows users, open up your **Command Prompt**.
Type in `java -version` and you will see the java version.

**Q**: How do I navigate to my folder in my terminal? <br>
**A**: For **Windows** user:
* Open Windows File Explorer.

> * Navigate to the folder that consist 

Reviewer: I think this one you will have to update with the details
1. Open Windows File Explorer.
2. Navigate to the parent folder of the target folder and double-click into it (the folder you want to open a command window into)
3. Click on the file explorer address bar of the folder you are in and clear the File Path (Eg. C:\Users\PC\Downloads\internBook) and type “cmd” and press enter.

Mac
1. Open Finder.
2. Navigate to the parent folder of the target folder but do not double-click into it (the folder you want to open a terminal window into)
3. Right-Click on the folder and hover over Services (At the bottom), Click New Terminal at Folder. 
    - If there no such option, you will have to go to your Settings 	> Keyboard > Keyboard Shortcuts.. > Services > Files
    and Folders > Tick New Terminal at Folder
4. Now try again
A Terminal at the folder location will pop up.

**Q**: Why is it possible to add applications with dates that have already passed?<br>
**A**: Users are permitted to add applications with past dates because we're in the process of developing new 
[features](#future-enhancements-hammer). These features will enable users to take notes about specific companies, which they can
refer back to when considering future applications.

**Q**: When is a company classified as a duplicate?<br>
**A**: A company is identified as a duplicate when you attempt to add a record with identical name, phone number, email 
address, tags, start date, and end date as an existing entry. If all these fields matches, the system flags it as a duplicate,
preventing the addition of such an application.

> **Q**: Why do applications with no dates appear between sorted applications by start or end date?

Reviewer 2: We can consider adding a `<br>` tag here

date entered, causing them not to display any dates. However, when sorting by start date, applications with only a start 
date will also be considered in the sorting process.

--------------------------------------------------------------------------------------------------------------------

## Known issues :construction:

1. **When using multiple screens**, if you move the application to a secondary screen, and later switch to using only the primary screen, the GUI will open off-screen. The remedy is to delete the `preferences.json` file created by the application before running the application again.
2. **Only when both start and end dates are present, will the date be shown in the GUI**. If only one of the dates is present, the sorts will take them into account while sorting, leading to misleading errors such as the one below.

![date sort](images/datesorterror.png)

3. **Only when both start and end dates are present, will the date be shown in the GUI**, this allows users to seemingly add in duplicate companies when in fact the start/end dates are different.

> 4. **Duplicate tags**, are allowed to be added into the same company, however they will still only show unique tags. 

Reviewer: for this maybe we can show an image to make it easier for the user to understand what is duplicate tags in the command
